### PR TITLE
Add `std::initializer_list` constructor for Dictionary.

### PR DESCRIPTION
**Pull request by bruvzg**
https://github.com/godotengine/godot-proposals/issues/1357 for `Dictionary`/`TypedDictionary`.

### CODE REVIEW

**Comment by Calinou** [comment]
I'm surprised it was this simple, considering https://github.com/godotengine/godot/pull/95920 had this to say on the matter:

> These exclusions make it impossible (or otherwise VERY tedious) to add initializer list support to Dictionaries as-is, given it internally relies on HashMap.

**Review by Repiteo — APPROVED** [review]
I'll be damned! Dunno how I ended up overlooking this method during my personal tests. #95920 should still happen eventually, but I guess it wasn't *needed* per se. Maybe I was thinking more of Iterators in saying that?

This implementation seems fine, though I'd wanna see a test on a typed dictionary as well

File: tests/core/variant/test_dictionary.h [hunk 1]
```diff
@@ -95,6 +95,15 @@ TEST_CASE("[Dictionary] Assignment using bracket notation ([])") {
...
 }
 
+TEST_CASE("[Dictionary] List init") {
+	Dictionary dict{ { 0, "int" }, { "key", 200 }, { Vector2(), "v2" } };
+
+	CHECK(dict.size() == 3);
+	CHECK(dict[0] == "int");
+	CHECK(dict["key"] == Variant(200));
+	CHECK(dict[Vector2()] == "v2");
+}
```
[the remark this answers is not in the document]
Comment (bruvzg) [reply tests/core/variant/test_dictionary.h:105]: > It's not implemented by https://github.com/godotengine/godot/pull/95920 either.

For Array it is Implemented by https://github.com/godotengine/godot/pull/89782 and https://github.com/godotengine/godot/pull/86015

**Comment by bruvzg** [comment]
Added the same for `TypedDictionary` as well.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**